src/ui/components/VPNCollapsibleCard.qml

```diff
@@ -0,0 +1,181 @@
+    width: parent.width
+
+    states: [
+        State {
+            name: "expanded"
+            when: root.expanded
+
+            PropertyChanges {
+                target: cardContent
+                height: cardContent.implicitHeight
+                opacity: 1
+            }
+
+            PropertyChanges {
+                target: chevron
+                rotation: -90
+            }
+        },
+        State {
+            name: "collapsed"
+            when: !root.expanded
+
+            PropertyChanges {
+                target: cardContent
+                height: 0
+                opacity: 0
+            }
+
+            PropertyChanges {
+                target: chevron
+                rotation: 90
+            }
+        }
+    ]
+
+    ColumnLayout {
+        id: cardWrapper
+
+        spacing: 0
+        width: root.width
+
+        // Card header
+        RowLayout {
+            id: accordionHeader
+
+            spacing: Theme.listSpacing * 2
+            Layout.topMargin: Theme.listSpacing
+            Layout.leftMargin: Theme.listSpacing * 2
+            Layout.rightMargin: Theme.listSpacing * 2
+            Layout.fillWidth: true
+
+            VPNIcon {
+                id: icon
+
+                source: iconSrc
+                visible: iconSrc !== ""
+            }
+
+            Text {
+                id: accordionTitle
+
+                Layout.fillWidth: true
+                Layout.preferredWidth: cardWrapper.width
+
+                color: Color.grey50
+                font.family: Theme.fontBoldFamily
+                font.pixelSize: Theme.fontSize
+                lineHeight: Theme.labelLineHeight
+                lineHeightMode: Text.FixedHeight
+                text: root.title
+                verticalAlignment: Text.AlignVCenter
+                wrapMode: Text.Wrap
+            }
+
+            Rectangle {
+                id: stateIndicator
+
+                color: accordionHeader.state === Theme.uiState.stateHovered || accordionHeader.state === Theme.uiState.statePressed
+                    ? Color.grey10
+                    : Color.white
+                height: Theme.rowHeight
+                radius: Theme.cornerRadius - 1
+                width: Theme.rowHeight
+
+                Behavior on color {
+                    ColorAnimation {
+                        duration: animationDuration
+                    }
+                }
+
+                VPNChevron {
+                    id: chevron
+
+                    isDark: true
+                    anchors {
+                        verticalCenter: parent.verticalCenter
+                        horizontalCenter: parent.horizontalCenter
+                    }
+
+                    Behavior on rotation {
+                        NumberAnimation {
+                            duration: animationDuration
+                            easing.type: Easing.OutQuad
+                        }
+                    }
+                }
+            }
```

Comment:
I did not use `VPNIconButton` here because of the rotation animation of the chevron. But now that I‘m revisiting this it maybe better to use a button item here accessibility reasons?

Reply:
Let's use `VPNIconButton` for this. I think a button is better for accessibility and we'll get all of the hover/focus/pressed/tooltip functionality baked into `VPNIconButton`/`VPNButtonBase` for free. We can nest `VPNChevron` inside and I think everything should still just work. 

Sidenote: I didn't know you could set anchor values like this!

 

```            
            VPNIconButton {
                id: stateIndicator
                Layout.preferredHeight: Theme.rowHeight
                Layout.preferredWidth: Theme.rowHeight
                accessibleName: "Need string: Expand card"
                onClicked: handleToggleCard()

                VPNChevron {
                    id: chevron

                    isDark: true
                    anchors.centerIn: parent

                    Behavior on rotation {
                        NumberAnimation {
                            duration: animationDuration
                            easing.type: Easing.OutQuad
                        }
                    }
                }
            }